plugins/outputs/iotdb/iotdb.go

```diff
@@ -0,0 +1,287 @@
+		s.TimeStampUnit == "millisecond" ||
+		s.TimeStampUnit == "microsecond" ||
+		s.TimeStampUnit == "nanosecond") {
+		errorMsg = fmt.Sprintf("IoTDB Config Warning: The value of 'TimeStampUnit' is invalid: %s. Now it's fixed to 'nanosecond'.", s.TimeStampUnit)
+		s.Log.Warnf(errorMsg)
+		s.TimeStampUnit = "nanosecond"
+	}
+	if !(s.TreateTagsAs == "Measurements" || s.TreateTagsAs == "DeviceID_subtree") {
+		errorMsg = fmt.Sprintf("IoTDB Config Warning: The value of 'TreateTagsAs' is invalid: %s. Now it's fixed to 'Measurements'.", s.TreateTagsAs)
+		s.Log.Warnf(errorMsg)
+		s.TreateTagsAs = "Measurements"
+	}
+	s.Log.Info("IoTDB output plugin initialization completed.")
+	return nil
+}
+
+func (s *IoTDB) Connect() error {
+	// Make any connection required here
+	config := &client.Config{
+		Host:     s.Host,
+		Port:     s.Port,
+		UserName: s.User,
+		Password: s.Password,
+	}
+	var ss = client.NewSession(config)
+	s.session = &ss
+	if err := s.session.Open(false, s.Timeout); err != nil {
+		s.Log.Errorf("IoTDB Connect Error: Fail to connect host:'%s', port:'%s', err:%v", s.Host, s.Port, err)
+		return err
+	}
+
+	return nil
+}
+
+func (s *IoTDB) Close() error {
+	// Close any connections here.
+	// Write will not be called once Close is called, so there is no need to synchronize.
+	_, err := s.session.Close()
+	if err != nil {
+		s.Log.Errorf("IoTDB Close Error: %v", err)
+	}
+	return nil
+}
+
+// Write should write immediately to the output, and not buffer writes
+// (Telegraf manages the buffer for you). Returning an error will fail this
+// batch of writes and the entire batch will be retried automatically.
+func (s *IoTDB) Write(metrics []telegraf.Metric) error {
+	// Convert Metrics to Records with Tags
+	recordsWithTags, err := s.ConvertMetricsToRecordsWithTags(metrics)
+	if err != nil {
+		s.Log.Errorf(err.Error())
+		return err
+	}
+	// Wirte to client
+	// status, err := s.session.InsertRecords(deviceidList, measurementsList, dataTypesList, valuesList, timestampList)
+	err = s.WriteRecordsWithTags(recordsWithTags)
+	if err != nil {
+		s.Log.Errorf("IoTDB Write Error: %s", err.Error())
+	}
+	return err
+}
+
+// Find out data type of the value and return it's id in TSDataType, and convert it if nessary.
+func (s *IoTDB) getDataTypeAndValue(value interface{}) (client.TSDataType, interface{}) {
+	switch v := value.(type) {
+	case int32:
+		return client.INT32, v
+	case int64:
+		return client.INT64, v
+	case uint32:
+		return client.INT64, int64(v)
+	case uint64:
+		if s.ConvertUint64To == "ToInt64" {
+			if v <= uint64(math.MaxInt64) {
+				return client.INT64, int64(v)
+			}
+			return client.INT64, int64(math.MaxInt64)
+		} else if s.ConvertUint64To == "ForceToInt64" {
+			return client.INT64, int64(v)
+		} else if s.ConvertUint64To == "Text" {
+			return client.TEXT, fmt.Sprintf("%d", v)
+		} else {
+			s.Log.Errorf("unknown converstaion configuration of 'convertUint64To': %s", s.ConvertUint64To)
+			return client.UNKNOW, int64(0)
+		}
```

Comment:
Use a switch statement here for readability!

Reply:
ok, it's my bad that I didn't use switch. Thanks for reminding me.